### ISSUE TITLE
scikit-image-contrib

### ISSUE THREAD

**Issue by emmanuelle** [scikit-image/scikit-image#4389]
It's been discussed several times to create a `scikit-image-contrib` (or a better name) repo and package, for high-quality contributions which are not of broad enough interest for scikit-image. An example is https://github.com/scikit-image/scikit-image/pull/4291.

Creating the repo is easy, but several questions need to be addressed.

- Structure of the package: same as scikit-image, with the same submodule names?
- CI: maybe more lightweight? 
- doc: landing page + gallery of examples? (so that it would still be required to have a gallery example when contributing a new function)
- release cycle/process?
- accepted dependencies? (needs to be pip/conda installable on Linux/Mac/Windows, but maybe we can accept more deps than scikit-image)

Etc. Please comment @skimage/core ! The burden of maintaining this package would probably be on us so we would have to find ways to make this more lightweight than for the main package, or it won't be sustainable.

**Comment by jni** [comment]
> Structure of the package

Yes, I think what makes the most sense here is to have a package `skimage_contrib` with all the same submodules, but added functions, e.g. `from skimage_contrib.filters import threshold_fancy_untested_method`.

> CI: maybe more lightweight?

Yes. I wouldn't mind if scikit-image-contrib was more forward-looking than scikit-image, e.g. only 1-2 Python versions, latest NumPy, Numba, etc, so that in CI we only really need to test with latest releases.

> doc: landing page + gallery of examples? (so that it would still be required to have a gallery example when contributing a new function)

Sure. I might be perfectly happy with just doctests, in some cases, e.g. thresholding.

> release cycle/process?

This might be the perfect testbed for the linear release process discussed in [zulip](https://skimage.zulipchat.com/#narrow/stream/181448-development/topic/release.20backports).

> accepted dependencies? (needs to be pip/conda installable on Linux/Mac/Windows, but maybe we can accept more deps than scikit-image)

Yes, as above.

imho the key here is to be much more lightweight in our requirements. As long as something follows our core API principles (PEP8 formatting, basic arrays as our first class object, etc.) and tests pass, it can go in.

**Comment by emmanuelle** [comment]
I created https://github.com/scikit-image/scikit-image-contrib

**Comment by sciunto** [comment]
@scikit-image/core (ping didn't work in the first post)

**Comment by alexdesiqueira** [comment]
Hey everyone,
@emmanuelle, the hypothesis package has something I really like in this sense, [extensions](https://hypothesis.readthedocs.io/en/latest/extras.html)! I thought on that before when we were discussing to separate the data from the main package (were we? I can't remember anymore). I guess we could have something related to that...?

**Comment by sciunto** [comment]
I started an issue to brainstorm on PRs that would be suitable for this repository: https://github.com/scikit-image/scikit-image-contrib/issues/2

I think it will help us also to define the perimeter.

**Comment by soupault** [comment]
Sorry for being late to the discussion. I am personally not very optimistic on sustainability of an auxiliary repo. What do you think about having a single codebase and doing the separation at a build-system level? The distributed format could then be either a extension or even a separate module.

This way it should be easier to keep the development, testing, documentation, and infrastructure manageable and organized.

**Comment by rfezzani** [comment]
Good suggestion @soupault, as I already proposed in #4081, should we consider using [namspace packaging](https://packaging.python.org/guides/packaging-namespace-packages/) for this purpose?

This packaging strategy may have some advantages:

- reduce the number of required dependencies by deporting "annex" sub-modules (`viewer`, `io`, `draw`, `data` for example),
- make a light 'core' package,
- ease the integration of new "annex"/"contrib"

But at the price of additional integration tests...

**Comment by stefanv** [comment]
Namespace packages (like skimage itself used to be, scikits.image), introduce its own whole series of problems. I'd probably steer clear, unless we've done thorough research to show that all problems have been addressed.

**Comment by alexdesiqueira** [comment]
My concerns are:
1. if we separate on another repo, we could risk having a "forgotten" part of skimage (was the repo supposed to be part of scikit-image?) — also, +1 on @soupault and @rfezzani pointed issues.
2. a namespace package could bloat the core package, I guess? Could you elaborate on some of the issues, @stefanv?
That's why I'm pointing a set of "extension" packages. Of course, I may be wrong — I don't really know how they work! Please point where my concerns fail y'all.
